src/admin/resolvers/mutations/ApprovedItem.ts

```diff
@@ -245,3 +258,108 @@ export async function uploadApprovedItemImage(
   const image = await data;
   return await uploadImageToS3(context.s3, image);
 }
+
+/**
+ * Imports an approved item
+ * Creates an approved item if it doesn't exist, and
+ * creates a scheduled item for the approved item
+ * if it doesn't exist
+ * @param parent
+ * @param data
+ * @param context
+ */
+export async function importApprovedItem(
+  parent,
+  { data },
+  context: IContext
+): Promise<ImportApprovedCuratedCorpusItemPayload> {
+  // check if user is authorized to import an item
+  if (!context.authenticatedUser.canWriteToCorpus()) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
+
+  // Check the language. Throw an error if not valid
+  checkLanguage(data.language);
+
+  let approvedItem = await getApprovedItemByUrl(context.db, data.url);
```

Comment:
i don't know how much it matters, but what do you think about inverting this logic? instead of checking existence up front (which might have a high performance cost?), we instead catch the already exists error and retrieve the existing id in the `catch` block? (same for scheduled item.)

Reply:
I think that makes sense - This is a new service so very high chance that all the records will not exist. In that case, attempting to create first, definitely saves us some database calls. 

But that also holds true for the opposite - what happens when we already have most of the records but need to rerun this again?

Reply:
i _think_ if we have to re-run a backfill, the assumption is we'd delete everything for that new tab first.